cpp/oneapi/dal/test/ccl_communicator.cpp

```diff
@@ -207,5 +218,60 @@ TEST_M(ccl_comm_test, "allgatherv") {
     }
 }
 
-} // namespace oneapi::dal::test
+TEST_M(ccl_comm_test, "allgatherv_empty_rank") {
+    auto comm = get_new_comm();
+    const std::int64_t granularity = 10;
+    const std::int64_t rank_count = comm.get_rank_count();
+    const std::int64_t rank = comm.get_rank();
+
+    std::vector<std::int64_t> recv_counts(rank_count);
+    std::vector<std::int64_t> displs(rank_count);
+    std::int64_t total_size = 0;
+    for (std::int64_t i = 0; i < rank_count; i++) {
+        recv_counts[i] = i != 1 ? (i + 1) * granularity : 0;
```

Comment:
It would be better to add a constant for empty rank - to make clear that "1" is a rank that should send nothing.